src/cmd_vel_mux.cpp

```diff
@@ -79,79 +104,109 @@ CmdVelMux::CmdVelMux(rclcpp::NodeOptions options) : rclcpp::Node("cmd_vel_mux",
+  std::map<std::string, ParameterValues>::const_iterator parameter;
+  std::map<std::string, std::shared_ptr<CmdVelSub>> new_map;
+
+  for (const std::pair<std::string, ParameterValues> & parameter : parameters)
   {
-    // Parse entries on YAML
-    std::string new_sub_name = names[i];
-    auto old_sub = std::find_if(list_.begin(), list_.end(),
-                                [&new_sub_name](const std::shared_ptr<CmdVelSub>& sub)
-                                                {return sub->name_ == new_sub_name;});
-    if (old_sub != list_.end())
+    std::string key = parameter.first;
+    ParameterValues parameter_values = parameter.second;
+    // Check if parameter subscriber has all its necessary values
+    if (map_.find(key) != map_.cend())
     {
-      // For names already in the subscribers list, retain current object so we don't re-subscribe to the topic
-      new_list[i] = *old_sub;
+      // For names already in the subscribers map, retain current object so we don't re-subscribe to the topic
+      new_map[key] = map_[key];
     }
     else
     {
-      new_list[i] = std::make_shared<CmdVelSub>();
+      new_map[key] = std::make_shared<CmdVelSub>();
     }
+
     // update existing or new object with the new configuration
 
-    // Fill attributes with a YAML node content
     double new_timeout;
     std::string new_topic;
-    new_list[i]->name_ = names[i];
-    new_topic = topics[i];
-    new_timeout = timeouts[i];
-    new_list[i]->priority_ = priorities[i];
-    new_list[i]->short_desc_ = short_descs[i];
+    new_map[key]->name_ = key;
+    new_topic = parameter_values.topic;
+    new_timeout = parameter_values.timeout;
+    new_map[key]->priority_ = parameter_values.priority;
+    new_map[key]->short_desc_ = parameter_values.short_desc;
 
-    if (new_topic != new_list[i]->topic_)
+    if (new_topic != new_map[key]->topic_)
     {
       // Shutdown the topic if the name has changed so it gets recreated on configuration reload
       // In the case of new subscribers, topic is empty and shutdown has just no effect
-      new_list[i]->topic_ = new_topic;
-      new_list[i]->sub_ = nullptr;
+      new_map[key]->topic_ = new_topic;
+      new_map[key]->sub_ = nullptr;
     }
 
-    if (new_timeout != new_list[i]->timeout_)
+    if (new_timeout != new_map[key]->timeout_)
     {
       // Change timer period if the timeout changed
-      new_list[i]->timeout_ = new_timeout;
-      new_list[i]->timer_ = nullptr;
+      new_map[key]->timeout_ = new_timeout;
+      new_map[key]->timer_ = nullptr;
     }
   }
 
-  list_ = new_list;
+  map_ = new_map;
 
   // (Re)create subscribers whose topic is invalid: new ones and those with changed names
   double longest_timeout = 0.0;
-  for (size_t i = 0; i < list_.size(); i++)
+  for (std::pair<const std::string, std::shared_ptr<CmdVelSub>> & m : map_)
   {
-    if (!list_[i]->sub_)
+    std::string key = m.first;
+    if (!m.second->sub_)
```

Comment:
@clalancette I'm inclined towards aliasing instead of using `std::pair` attributes. It does not help readability.